nemo/collections/nlp/modules/common/text_generation_utils.py

```diff
@@ -444,11 +545,19 @@ def switch(val1, val2, boolean):
 
 def forward_step(model, batch, tensor_shape):
 
+    # Should call MegatronGPTPPromptLearningModel's forward method
+    if hasattr(model, 'prompt_table'):
+        forward_model = model
```

Comment:
Nice trick to make prompt model work in the generation code.
Instead of test it has attribute. It is less hacky to do type check
```python
if isinstance(model,  MegatronGPTPPromptLearningModel):
```

Reply:
I originally wanted to do what you suggested, but it resulted in a circular import because the `megatron_gpt_prompt_learning_model.py` already imports `text_generation_utils.py`

Reply:
yeah. cyclic import is an issue. one solution is not importing it at the beginning 
do something like
```python
if isinstance(model, nemo.collections.nlp.language_modeling.megatron_gpt_prompt_learning_model.MegatronGPTPPromptLearningModel):```
or import it locally inside the function

Reply:
This is kind of long, but I agree its less hacky. I'll add it.